SU-HSDO/suhumsci
# STN-675: Add Gradient Hero Styles 🎨

# STN-657: Gradient Hero Styles READY FOR REVIEW

## Summary
The work in this PR focuses on adding styles to the new gradient hero paragraph type. This work has been branched off of the gradient hero configuration / functionality in #841.

This PR should not be reviewed or merged until after #841!

## Testing / Tasks completed
- [x] Animation
- [x] Dark Color Inversion
- [x] Browsers
- [x] Logged in / out
- [x] Default / Right Aligned
- [x] Colorful / Traditional
- [x] Captions/Credits
- [x] Zoom in 200% / Zoom out 80%
- [n/a] Utility Classes - this work does not touch any utility classes
- [n/a] Color Pairings - this work is the same across all color pairings
- [x] Updated Color Pairing Spreadsheet

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In your CLI, run npm run test and confirm all tests pass.
2. To test a gradient, create a new flexible page or open an existing one. Under Components add a Gradient Hero. Fill in all the fields.
3. By default, the gradient hero should not be available via the components dropdown when creating a flexible page. In order to test the work in this PR you will need to go to the Component Settings for a Flexible Page and `uncheck` the `Gradient Hero` under `Reference Type`. Be sure to revert this change when you are done testing the scenarios listed below. Quick link:  http://sparkbox-sandbox.suhumsci.loc/admin/structure/types/manage/hs_basic_page/fields/node.hs_basic_page.field_hs_page_components
4. By default, the gradient hero should also not be available via the row component dropdown. In order to test the gradient hero within a row, you will need to go to the Component Settings for a Flexible Page for Rows and `uncheck` the `Gradient Hero` under `Reference Type`. Again be sure to revert this change when testing is complete. Quick link: http://sparkbox-sandbox.suhumsci.loc/admin/structure/paragraphs_type/hs_row/fields
5. To test the gradient hero in the hero section of a flexible page, you must add it to the hero image settings configuration. Under this quick link: http://sparkbox-sandbox.suhumsci.loc/admin/structure/types/manage/hs_basic_page/fields/node.hs_basic_page.field_hs_page_hero `check` the `Gradient Hero` under `Reference Type`. Remember to revert this change when testing is complete.

Test the following scenarios:
- [x] A gradient hero within the hero component section at the top of a page.
- [x] A gradient hero within the main content section of a page.
- [x] A gradient hero with default alignment to the left.
- [ ] A gradient hero where the alignment has been changed to `-None-` also has the `hb-hero-gradient--content-default` class.
- [x] A gradient hero where the alignment has been set to `Right Aligned`
- [x] Add a caption / credit to a gradient hero image. It should display the correct caption / credit styles (white text on a black background that fades to transparent.)
- [x] The gradient hero is keyboard accessible.
- [x] The gradient hero can be added to the Colorful and Traditional themes.

6. The work in this PR includes a refactor of common hero styles into several mixins. Perform a visual review of the hero image overlay, spotlight, carousel, and banner components to ensure all styles render as expected and that the refactor did not unintentionally cause any regressions.
